### PR TITLE
Add actor table to global state API

### PR DESCRIPTION
**Pull request by pcmoritz**
## Why are these changes needed?

This introduces an actor table to the GCS python interface.

It is needed to make sure the Ray Dashboard gets already existing actors after subscribing to the actor channel, to avoid a race condition.

The PR also reintroduces some tests that were deactivated when direct calls were switched on by default and fixes a bug where the WebUI was an additional driver in the ray cluster.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: python/ray/state.py [hunk 1]
```diff
@@ -1120,6 +1185,19 @@ def node_ids():
     return node_ids
 
 
+def actors(actor_id=None):
+    """Fetch and parse the actor table info for one or more actor IDs.
```
Comment (robertnishihara) [inline python/ray/state.py]: I realize that we do this in some of the other methods, but please don't use the term "actor table" in user-facing documentation. The idea of a "table" is confusing and really more of an implementation detail. The documentation should just use concepts that users think about like "actors".

Comment (robertnishihara) [reply python/ray/state.py]: This applies in the args also

File: python/ray/tests/test_advanced_3.py [hunk 2]
```diff
@@ -89,6 +89,15 @@ def f(x):
     attempt_to_load_balance(f, [x], 100, num_nodes, 25)
 
 
+def wait_for_num_actors(num_actors, timeout=10):
```
Comment (robertnishihara) [inline python/ray/tests/test_advanced_3.py]: Could define this inline since it's only used once.

File: python/ray/state.py [hunk 3]
```diff
@@ -306,6 +306,71 @@ def object_table(self, object_id=None):
...
+        gcs_entries = gcs_utils.GcsEntry.FromString(message)
+
+        assert len(gcs_entries.entries) == 1
+        actor_table_data = gcs_utils.ActorTableData.FromString(
+            gcs_entries.entries[0])
+
+        actor_info = {
+            "JobID": binary_to_hex(actor_table_data.job_id),
+            "Address": {
+                "IpAddress": actor_table_data.address.ip_address,
```
Comment (robertnishihara) [inline python/ray/state.py]: This capitalization is really weird. It should probably be `"IP Address"`.

Comment (robertnishihara) [reply python/ray/state.py]: same below. If you really don't like the space, then `"IPAddress"`

File: python/ray/tests/test_advanced_3.py (hunk 2 above)
Comment (pcmoritz) [reply python/ray/tests/test_advanced_3.py]: Given that the others are also outside, it is probably better to keep it consistent :)

**Review by robertnishihara — APPROVED** [review]

**Pull request merged**